x/evm/keeper/sdk_tx.go

```diff
@@ -0,0 +1,121 @@
+package keeper
+
+import (
+	"fmt"
+	"math"
+
+	sdkmath "cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/evmos/ethermint/x/evm/types"
+)
+
+// CheckSDKTxFee implements the `TxFeeChecker` for ante handler,
+// could be called in both checkTx and deliverTx modes.
```

Comment:
imo this name is not very descriptive. Also, the logic is not well documented here, what does this function does? The comment only mentions where it's used

Reply:
done, added more comments.